hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterAdminServer.java

```diff
@@ -358,6 +360,25 @@ public AddMountTableEntryResponse addMountTableEntry(
     return getMountTableStore().addMountTableEntry(request);
   }
 
+  @Override
+  public AddMountTableEntriesResponse addMountTableEntries(AddMountTableEntriesRequest request)
+      throws IOException {
+    List<MountTable> mountTables = request.getEntries();
+    for (MountTable mountTable : mountTables) {
+      verifyMaxComponentLength(mountTable);
+    }
+    if (this.mountTableCheckDestination) {
+      for (MountTable mountTable : mountTables) {
+        List<String> nsIds = verifyFileInDestinations(mountTable);
+        if (!nsIds.isEmpty()) {
+          throw new IllegalArgumentException(
+              "File not found in downstream " + "nameservices: " + StringUtils.join(",", nsIds));
+        }
```

Comment:
This code is now duplicated in three places. Can we move it into a small separate function.